linera-views/src/register_view.rs

```diff
@@ -58,29 +62,45 @@ where
     }
 
     fn rollback(&mut self) {
+        self.was_cleared = false;
+        self.after_clear_set = false;
         self.update = None;
         *self.hash.get_mut() = self.stored_hash;
     }
 
     fn flush(&mut self, batch: &mut Batch) -> Result<(), ViewError> {
-        if let Some(value) = self.update.take() {
-            let key = self.context.base_tag(KeyTag::Value as u8);
-            batch.put_key_value(key, &value)?;
-            self.stored_value = value;
+        if self.was_cleared {
+            batch.delete_key_prefix(self.context.base_key());
+        }
+        // If there is clearing then we want to write the value only if it has been set afterward
+        if !self.was_cleared || self.after_clear_set {
```

Comment:
This test would disappear if we set `self.update = None` in `clear()`.

Reply:
This has been simplified by other corrections by Andreas.